src/CommunityToolkit.Maui.SourceGenerators/Extensions/NamedTypeSymbolExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Extensions;
+
+static class NamedTypeSymbolExtensions
+{
+	public static bool ImplementsInterfaceOrBaseClass(this INamedTypeSymbol typeSymbol, in string typeToCheck)
+	{
+		if (typeSymbol is null)
+		{
+			return false;
+		}
+
+		if (typeSymbol.MetadataName == typeToCheck)
+		{
+			return true;
+		}
+
+		if (typeSymbol.BaseType?.MetadataName == typeToCheck)
+		{
+			return true;
+		}
```

Comment:
Is it intended this doesn't go through all base types?

Reply:
We aren't using this method anymore, I'll remove it in a future commit